app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -668,11 +680,8 @@ class RefreshController @Inject constructor(
         }
     }
 
-    private fun MutableRealm.addNewThreadToFolder(newThread: Thread, folder: Folder, impactedThreadsManaged: MutableSet<Thread>) {
-        ThreadController.upsertThread(newThread, realm = this).also {
-            folder.threads.add(it)
-            impactedThreadsManaged += it
-        }
+    private fun MutableRealm.putNewThreadInRealm(newThread: Thread, impactedThreadsManaged: MutableSet<Thread>) {
+        impactedThreadsManaged += ThreadController.upsertThread(newThread, realm = this)
```

Comment:
Does this method not not hide the mutation of `impactedThreadsManaged` a bit too much?